File: app/src/main/java/com/mycompany/myapp/util/RxUtils.java
package com.mycompany.myapp.util;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class RxUtils {

    public static void disposeIfNotNull(Disposable disposable) {
        if (disposable != null) {
            disposable.dispose();
        }
    }

    public static CompositeDisposable getNewCompositeDisposableIfDisposed(CompositeDisposable disposable) {
        if (disposable == null || disposable.isDisposed()) {
            return new CompositeDisposable();
        }

        return disposable;
    }
}

File: app/src/main/java/com/mycompany/myapp/util/RxUtils.kt
package com.mycompany.myapp.util

import io.reactivex.disposables.CompositeDisposable

object RxUtils {
    fun getSafeCompositeDisposable(disposable: CompositeDisposable?): CompositeDisposable {
        return when {
            disposable == null || disposable.isDisposed -> CompositeDisposable()
            else -> disposable
        }
    }
}